de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/controller/ProjectExplorerToolBarController.java

```diff
@@ -0,0 +1,192 @@
+	 */
+	public ProjectExplorerToolBarController(ToolBar toolbar, IEclipseContext context, ServiceContainer services,
+			Shell shell) {
+		bar = toolbar;
+		this.services = services;
+
+		// Search Section
+		TextField search = new TextField();
+		search.setVisible(false);
+		search.setOnAction(actionEvent -> {
+			services.eventBroker.send(E4CEventTable.EVENT_FILTER_CHANGED, search.getText());
+		});
+
+		ImageView searchImage = services.imageService.getFXImage(null, FileTable.SEARCH_PNG);
+		Tooltip searchTooltip = new Tooltip("Search Files");
+		ImageView closeSearchImage = services.imageService.getFXImage(null, FileTable.X_PNG);
+		Tooltip closeSearchTooltip = new Tooltip("Exit Search");
+		Button searchButton = new Button("", searchImage);
+		searchButton.setTooltip(searchTooltip);
+		searchButton.setOnAction(actionEvent -> {
+			search.setVisible(!search.isVisible());
+			// Removing and adding the search bar allows automatic resizing
+			if (search.isVisible()) {
+				searchButton.setGraphic(closeSearchImage);
+				searchButton.setTooltip(closeSearchTooltip);
+				bar.getItems().add(1, search);
+				search.requestFocus();
+			} else {
+				searchButton.setGraphic(searchImage);
+				searchButton.setTooltip(searchTooltip);
+				search.setText("");
+				bar.getItems().remove(search);
+				services.eventBroker.send(E4CEventTable.EVENT_FILTER_CHANGED, null);
+			}
+
+		});
+		bar.getItems().add(searchButton);
+
+		bar.getItems().add(new Separator());
+
+		// New Folder
+		btnNewFolder = createToolbarButton("Create New Folder", FileTable.NEWFOLDER_PNG, actionEvent -> {
+			NewFolderHandler handler = new NewFolderHandler();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
+
+		// File Actions
+		btnNewFile = createToolbarButton("Create new File", FileTable.NEWFILE_PNG, actionEvent -> {
+			NewFileHandler handler = new NewFileHandler();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
+
+		// Import Files
+		btnImportFiles = createToolbarButton("Import Files", FileTable.FILE_PNG, actionEvent -> {
+			FileImportHandler handler = new FileImportHandler();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
+
+		// Delete
+		btnDelete = createToolbarButton("Delete Selected Entries", FileTable.DELETE_PNG, actionEvent -> {
+			RemoveFileCommand handler = new RemoveFileCommand();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
+
+		// Show in Explorer
+		btnShowExplorer = createToolbarButton("Show in Explorer", FileTable.EXPLORER_PNG, actionEvent -> {
+			ShowInExplorerHandler handler = new ShowInExplorerHandler();
+			if (handler.canExecute(services.rcpSelectionService)) {
+				handler.execute(context, services, shell);
+			}
+		});
```

Comment:
Ich glaube die ganzen Buttons in der Toolbar würde ich im FXML beschreiben und gucken das die actions als methoden im code stehen.